lib/rules/no-observers.js

```diff
@@ -30,40 +31,67 @@ module.exports = {
       context.report(node, ERROR_MESSAGE);
     };
 
+    let importedEmberName = undefined;
+    let importedObserverName = undefined;
+    let importedObservesName = undefined;
+    let importedAddObserverName = undefined;
+
     return {
       CallExpression(node) {
-        if (ember.isModule(node, 'observer')) {
-          report(node);
-        }
-
         const { callee } = node;
-
         if (
-          (types.isThisExpression(callee.object) || types.isMemberExpression(callee)) &&
+          // observer();
+          types.isIdentifier(callee) &&
+          callee.name === importedObserverName
+        ) {
+          report(node);
+        } else if (
+          // Ember.observer();
+          types.isMemberExpression(callee) &&
+          types.isIdentifier(callee.object) &&
+          callee.object.name === importedEmberName &&
+          types.isIdentifier(callee.property) &&
+          callee.property.name === 'observer'
+        ) {
+          report(node);
+        } else if (
+          // this.addObserver();
+          // Ember.addObserver();
+          // foo.addObserver();
+          types.isMemberExpression(callee) &&
           types.isIdentifier(callee.property) &&
           callee.property.name === 'addObserver'
         ) {
           report(node);
+        } else if (
+          // addObserver();
+          types.isIdentifier(callee) &&
+          callee.name === importedAddObserverName
+        ) {
+          report(node);
         }
       },
 
       Decorator(node) {
-        if (ember.isObserverDecorator(node)) {
+        if (ember.isObserverDecorator(node, importedObservesName)) {
           report(node);
         }
       },
 
       ImportDeclaration(node) {
-        const importSource = node.source.value;
-
-        if (
-          (importSource === '@ember/object' &&
-            node.specifiers.some(
-              (s) => (s.imported ? s.imported.name : s.local.name) === 'observer'
-            )) ||
-          importSource === '@ember/object/observers'
-        ) {
-          report(node);
+        if (node.source.value === 'ember') {
```

Comment:
I remember before we discussed making a more general solution to aggregating import information, so that we wouldn't need to keep reimplementing this in every `ImportDeclaration`. I think it would help a lot, since we keep doing this.

Reply:
Very interested in that change still. I started on it in #910 but it turned out to be too big of a change for me to take on at the time. Would love it if someone else was interested to pursue that!